hack/verify-reports.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+error() {
+  echo "ERROR: $*" 1>&2
+}
+
+info() {
+  echo "INFO: $*" 1>&2
+}
+
+# Check if the provided Gateway API version is greater than or equal to v1.1.0
+check_ge_v1.1.0() {
+    local version=$1
+    local minimum_version="v1.1.0"
+
+    # Normalize versions to remove leading 'v' and compare using sort -V
+    if [[ $(echo -e "${version#v}\n${minimum_version#v}" | sort -V | head -n1) == "${minimum_version#v}" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Check if the report fields are valid according to the rules defined in https://github.com/kubernetes-sigs/gateway-api/blob/release-1.1/conformance/reports/README.md
+check_report_fields() {
+    local report=$1
+    local expected_gateway_api_version=$2
+
+    # Check if the implementation version is a valid semver
+    local version=$(yq eval '.implementation.version' "$report")
+    local gateway_api_version=$(yq eval '.gatewayAPIVersion' "$report")
+    local gateway_api_channel=$(yq eval '.gatewayAPIChannel' "$report")
+    local mode=$(yq eval '.mode' "$report")
+    local version_pattern="^v?[0-9]+\.[0-9]+(\.[0-9]+)?$"
+
+    if [[ ! $version =~ $version_pattern ]]; then
```

Comment:
yeah, I see. But I think we should ensure the version here is not a branch. What about allowing
- semantic version
- pseudo-version
- commit hash

What else should we allow?

Reply:
I'm not exactly sure how we'll represent this for GKE, since it's not really tied to any customer facing versions. In general, I think I'd rather just rely on written guidelines and PR review.

Reply:
Yep, actually the GKE versioning in the conformance report PR does not follow such a pattern. I removed this check.